stored_responses/stored_responses.go

```diff
@@ -75,53 +74,62 @@ func extractStoredResponsesIds(impInfo []ImpExtPrebidData,
 	impBidderReplaceImp := ImpBidderReplaceImpID{}
 
 	for index, impData := range impInfo {
-		impId, err := jsonparser.GetString(impData.Imp, "id")
+		impId := impData.ID
+		impExt, err := impData.GetImpExt()
 		if err != nil {
-			return nil, nil, nil, nil, fmt.Errorf("request.imp[%d] missing required field: \"id\"", index)
+			return nil, nil, nil, nil, err
+		}
+		impExtPrebid := impExt.GetPrebid()
+		if impExtPrebid == nil {
+			continue
 		}
 
-		if impData.ImpExtPrebid.StoredAuctionResponse != nil {
-			if len(impData.ImpExtPrebid.StoredAuctionResponse.ID) == 0 {
+		if impExtPrebid.StoredAuctionResponse != nil {
+			if len(impExtPrebid.StoredAuctionResponse.ID) == 0 {
 				return nil, nil, nil, nil, fmt.Errorf("request.imp[%d] has ext.prebid.storedauctionresponse specified, but \"id\" field is missing ", index)
 			}
-			allStoredResponseIDs = append(allStoredResponseIDs, impData.ImpExtPrebid.StoredAuctionResponse.ID)
+			allStoredResponseIDs = append(allStoredResponseIDs, impExtPrebid.StoredAuctionResponse.ID)
 
-			impAuctionResponseIDs[impId] = impData.ImpExtPrebid.StoredAuctionResponse.ID
+			impAuctionResponseIDs[impId] = impExtPrebid.StoredAuctionResponse.ID
 
 		}
-		if len(impData.ImpExtPrebid.StoredBidResponse) > 0 {
+		if len(impExtPrebid.StoredBidResponse) > 0 {
+
+			// bidders can be specified in imp.ext and in imp.ext.prebid.bidders
+			allBidderNames := make([]string, 0)
+			for bidderName := range impExtPrebid.Bidder {
+				allBidderNames = append(allBidderNames, bidderName)
+			}
+			for extData := range impExt.GetExt() {
+				// no bidders will not be processed
+				allBidderNames = append(allBidderNames, extData)
+			}
 
 			bidderStoredRespId := make(map[string]string)
 			bidderReplaceImpId := make(map[string]bool)
-			for _, bidderResp := range impData.ImpExtPrebid.StoredBidResponse {
+			for _, bidderResp := range impExtPrebid.StoredBidResponse {
 				if len(bidderResp.ID) == 0 || len(bidderResp.Bidder) == 0 {
 					return nil, nil, nil, nil, fmt.Errorf("request.imp[%d] has ext.prebid.storedbidresponse specified, but \"id\" or/and \"bidder\" fields are missing ", index)
 				}
-				bidderName := bidderResp.Bidder
-				normalizedCoreBidder, ok := openrtb_ext.NormalizeBidderName(bidderResp.Bidder)
-				if ok {
-					bidderName = normalizedCoreBidder.String()
-				}
-				//check if bidder is valid/exists
-				if _, isValid := bidderMap[bidderName]; !isValid {
-					return nil, nil, nil, nil, fmt.Errorf("request.imp[impId: %s].ext.prebid.bidder contains unknown bidder: %s. Did you forget an alias in request.ext.prebid.aliases?", impId, bidderResp.Bidder)
-				}
```

Comment:
I see that this logic was removed. Isn't this logic checking that the stored response bidder name is the name of a bidder defined in the imp? Do we need this? Does this error detection exist elsewhere?